scripts/release.js

```diff
@@ -1,211 +1,167 @@
+
+  process.chdir(path.join(__dirname, '..'));
+  spawnOrFail('rm', ['-rf ../amazon-chime-sdk']);
+};
+
+const getCurrentRemoteBranch = () => {
+  return (spawnOrFail('git', ['for-each-ref --format="%(upstream:short)" "$(git symbolic-ref -q HEAD)"'], { skipOutput: true })).trim();
+};
+
+const buildAndPack = () => {
+  logger.log('Building package...');
+  spawnOrFail('npm', [`run build:publish`]);
+  logger.log('Packaging ...');
+  spawnOrFail('npm', ['pack'], { printErr: true });
+};
+
+const updateJSSdkDep = () => {
   // Update the peer dependency to the most updated version of the SDK
-  logger.log(`Installing SDK Version: ${updatedSdkVersion} as a peerDependency and devDependency of the react library.`);
+  logger.log(`Installing SDK Version: ${latestNPMJSSdkVersion} as a peerDependency and devDependency of the react library.`);
 
   let componentsPackageJson = JSON.parse(fs.readFileSync('package.json', 'utf-8'));
-  componentsPackageJson.peerDependencies['amazon-chime-sdk-js'] = `^${updatedSdkVersion}`;
-  componentsPackageJson.devDependencies['amazon-chime-sdk-js'] = `^${updatedSdkVersion}`;
+  componentsPackageJson.peerDependencies['amazon-chime-sdk-js'] = `^${latestNPMJSSdkVersion}`;
+  componentsPackageJson.devDependencies['amazon-chime-sdk-js'] = `^${latestNPMJSSdkVersion}`;
 
   fs.writeFileSync(
     'package.json',
     JSON.stringify(componentsPackageJson, null, 2)
   );
 
-  logger.log("NPM Installing component library...")
   spawnOrFail('npm', [`install`]);
-
+};
+
+const checkNPMDepsInstall = () => {
+  // Creates a new React test app, install component and sdk and build the test app to ensure no peer dependency warnings, errors or build issues
+  logger.log('Install amazon-chime-sdk-component-library-react and amazon-chime-sdk-js sdk as dependencies, check if there is peer dependency warning for amazon-chime-sdk-component-library-react');
+  process.chdir(path.join(__dirname, '../..'));
+  if (!fs.existsSync('dependency-check-app')){
+    spawnOrFail('mkdir', ['dependency-check-app']);
+  } else {
+    spawnOrFail('rm', ['-rf dependency-check-app']);
+    spawnOrFail('mkdir', ['dependency-check-app']);
+  }
+  process.chdir(path.join(__dirname, '../../dependency-check-app'));
+  spawnOrFail('npm', ['init -y']);
+  spawnOrFail('npm', ['install react react-dom']);
+  spawnOrFail('npm', [`install amazon-chime-sdk-js@${latestNPMJSSdkVersion} styled-components styled-system`]);
+  checkWarning('npm', [`install -q ../amazon-chime-sdk-component-library-react/amazon-chime-sdk-component-library-react-${currentVersion}.tgz`], null, 'amazon-chime-sdk-component-library-react');
   process.chdir(path.join(__dirname, '..'));
+  spawnOrFail('rm', ['-rf ../dependency-check-app']);
 }
 
-spawnOrFail('git', ['add -A']);
-spawnOrFail('git', [`commit -m 'Publish ${tag}'`]);
-logger.log(`Created commit called: "Publish ${tag}"`);
+const cleanUp = (remoteBranch) => {
```

Comment:
resetToHead

Reply:
As mention in the other PR in JS SDK, i want to keep this function generic as it can be whatever and not just reset to HEAD (we also have another command to clear out untracked changes too).